### PR TITLE
Added demo for filter component

### PR DESCRIPTION
**Pull request by Blackbaud-AlexKingman**

### CODE REVIEW

File: src/app/docs/filter/demo/inline-filter-demo.component.html [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      >
+      </sky-filter-button>
+    </sky-toolbar-item>
+  </sky-toolbar-section>
+</sky-toolbar>
+<sky-filter-inline
+  [hidden]="!showInlineFilters"
+>
+  <sky-filter-inline-item>
+    <label
```
Comment (blackbaud-johnly) [inline src/app/docs/filter/demo/inline-filter-demo.component.html:17]: Should this label have the `sky-control-label` class? It's really tight on the select field, so I wonder if the class would fix the spacing.

File: src/app/docs/filter/demo/modal-filter-demo-modal.component.html [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      (click)="applyFilters()"
+    >
+      Apply filters
+    </button>
+    <button
+      class="sky-btn sky-btn-link"
+      type="button"
+      (click)="clearAllFilters()"
+    >
+      Clear all filters
```
Comment (blackbaud-johnly) [inline src/app/docs/filter/demo/modal-filter-demo-modal.component.html:49]:
```suggestion
      Clear filters
```

File: src/app/public/plugin-resources/code-examples/filter/modal/filter-demo-modal.component.html [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      (click)="applyFilters()"
+    >
+      Apply filters
+    </button>
+    <button
+      class="sky-btn sky-btn-link"
+      type="button"
+      (click)="clearAllFilters()"
+    >
+      Clear all filters
```
Comment (blackbaud-johnly) [inline src/app/public/plugin-resources/code-examples/filter/modal/filter-demo-modal.component.html:49]:
```suggestion
      Clear filters
```

File: src/app/public/plugin-resources/code-examples/filter/inline/filter-demo.component.ts [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+  public filtersActive: boolean = false;
+
+  public fruitType: string = 'any';
+
+  public hideOrange: boolean = false;
+
+  public items: any[] = [
+      {
+          name: 'Orange',
+          description: 'A round, orange fruit.',
```
[the remark this answers is not in the document]
Comment (Blackbaud-AlexKingman) [reply src/app/public/plugin-resources/code-examples/filter/inline/filter-demo.component.ts:24]: Good catch. Removed.

File: src/app/docs/filter/demo/inline-filter-demo.component.html (hunk 1 above)
Comment (Blackbaud-AlexKingman) [reply src/app/docs/filter/demo/inline-filter-demo.component.html:17]: Yeah it is a little tight. I had to add a margin class to it as well. Good suggestion.

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-lists?_sp.spa=pr-filter-demo](https://developer.blackbaud.com/skyux-lists?_sp.spa=pr-filter-demo)

File: src/app/docs/filter/demo/inline-filter-demo.component.html (hunk 1 above)
Comment (blackbaud-johnly) [reply src/app/docs/filter/demo/inline-filter-demo.component.html:17]: Not sure if it's just me, but the label still seems tight to the input:
image
This is a minor cosmetic issue though, so I'll go ahead and approve. If you think we can tweak something to resolve this, I'm all for it. but we can always just circle back to this later.

**Review by blackbaud-johnly — APPROVED** [review]

File: src/app/docs/filter/demo/modal-filter-demo-modal.component.html [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  <sky-modal-header>
+    Food preferences
+  </sky-modal-header>
+  <sky-modal-content>
+    <label for="sky-demo-select-type">
+      Fruit type
+    </label>
+    <select
+      class="sky-form-control"
+      id="sky-demo-select-type"
```
Comment (Blackbaud-SteveBrush) [inline src/app/docs/filter/demo/modal-filter-demo-modal.component.html:11]: Should we use `skyId` here?

File: src/app/public/plugin-resources/code-examples/filter/inline/filter-demo.component.html [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+  [hidden]="!showInlineFilters"
+>
+  <sky-filter-inline-item>
+    <label
+      for="sky-demo-select-type"
+    >
+      Fruit type
+    </label>
+    <select
+      id="sky-demo-select-type"
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/plugin-resources/code-examples/filter/inline/filter-demo.component.html:23]: `skyId`?

File: src/app/public/plugin-resources/code-examples/filter/inline/filter-demo.component.ts [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+  public filteredItems: any[];
+
+  public filtersActive: boolean = false;
+
+  public fruitType: string = 'any';
+
+  public hideOrange: boolean = false;
+
+  public items: any[] = [
+      {
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/plugin-resources/code-examples/filter/inline/filter-demo.component.ts:22]: One too many tabs.

File: src/app/public/plugin-resources/code-examples/filter/modal/filter-demo-modal.component.html [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+  <sky-modal-header>
+    Food preferences
+  </sky-modal-header>
+  <sky-modal-content>
+    <label for="sky-demo-select-type">
+      Fruit type
+    </label>
+    <select
+      class="sky-form-control"
+      id="sky-demo-select-type"
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/plugin-resources/code-examples/filter/modal/filter-demo-modal.component.html:11]: `skyId`?

File: src/app/docs/filter/demo/inline-filter-demo.component.html (hunk 1 above)
Comment (Blackbaud-AlexKingman) [reply src/app/docs/filter/demo/inline-filter-demo.component.html:17]: Good catch. That label was missing a class. Fixed!

File: src/app/docs/filter/demo/modal-filter-demo-modal.component.html (hunk 5 above)
Comment (Blackbaud-AlexKingman) [reply src/app/docs/filter/demo/modal-filter-demo-modal.component.html:11]: Yes! Done.

File: src/app/public/plugin-resources/code-examples/filter/inline/filter-demo.component.html (hunk 6 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/plugin-resources/code-examples/filter/inline/filter-demo.component.html:23]: Yep. Done.

File: src/app/public/plugin-resources/code-examples/filter/inline/filter-demo.component.ts (hunk 7 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/plugin-resources/code-examples/filter/inline/filter-demo.component.ts:22]: Fixed.

File: src/app/public/plugin-resources/code-examples/filter/modal/filter-demo-modal.component.html (hunk 8 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/plugin-resources/code-examples/filter/modal/filter-demo-modal.component.html:11]: Done.

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-lists?_sp.spa=pr-filter-demo](https://developer.blackbaud.com/skyux-lists?_sp.spa=pr-filter-demo)

File: src/app/docs/filter/demo/inline-filter-demo.component.html [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+      class="sky-control-label sky-margin-inline-default"
+      [for]="fruitInput.id"
+    >
+      Fruit type
+    </label>
+    <select
+      id="sky-demo-select-type"
```
Comment (Blackbaud-SteveBrush) [inline src/app/docs/filter/demo/inline-filter-demo.component.html:24]: Might need to remove the `id` attribute now that you're using `skyId`.

File: src/app/public/plugin-resources/code-examples/filter/inline/filter-demo.component.html [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+>
+  <sky-filter-inline-item>
+    <label
+      class="sky-control-label sky-margin-inline-default"
+      [for]="fruitInput.id"
+    >
+      Fruit type
+    </label>
+    <select
+      id="sky-demo-select-type"
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/plugin-resources/code-examples/filter/inline/filter-demo.component.html:24]: Might need to remove the `id` attribute now that you're using `skyId`.

File: src/app/public/plugin-resources/code-examples/filter/modal/filter-demo-modal.component.html [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+  <sky-modal-content>
+    <label
+      class="sky-control-label"
+      [for]="fruitInput.id"
+    >
+      Fruit type
+    </label>
+    <select
+      class="sky-form-control"
+      id="sky-demo-select-type"
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/plugin-resources/code-examples/filter/modal/filter-demo-modal.component.html:14]: Might need to remove the `id` attribute now that you're using `skyId`.

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-lists?_sp.spa=pr-filter-demo](https://developer.blackbaud.com/skyux-lists?_sp.spa=pr-filter-demo)

**Review by blackbaud-johnly — APPROVED** [review]

**Pull request merged**